pkg/rebuild/maven/infer.go

```diff
@@ -283,7 +283,7 @@ func findPomXML(commit *object.Commit, pkg string) (*PomXML, string, error) {
 	var names []string
 	var pomXMLs []PomXML
 	commitTree.Files().ForEach(func(f *object.File) error {
-		if !strings.HasSuffix(f.Name, "pom.xml") {
+		if filepath.Base(f.Name) != "pom.xml" || strings.Contains(f.Name, "test/") {
```

Comment:
Hmm I don't know about the test constraint. Can test appear in a genuine path? Is this the differentiator that unlocks the 4 entries in the benchmark?

Reply:
Uh maybe we could pick the shortest then? Either by character length or path component length? I might favor the latter